src/client/extension.ts

```diff
@@ -109,14 +109,15 @@ function startREPLProc(context: ExtensionContext, config: settings.ISettings, pw
             let filePath = fileURLToPath(document.uri)
             proc.log.appendLine(`executing: ${filePath}`)
 
```

Comment:
What would happen if the buffer is dirty?
Shall we write it to the disk for the user?

Reply:
Do you mean, if they've already typed something into the prompt? 

That's handled by PowerShellEditorServices.

Reply:
I think it means if I'm editing `test.ps1` and I've changed the file without saving it.

In that case, I think writing to disk is the thing to do, that's the behavior in both VS Code and the ISE (although the ISE prompts you that it will be saving it...)

Reply:
ah so save it before running it?

Reply:
Yeah probably. The only question might be if it's an untitled document...

Reply:
Good point… maybe for untitled we just run the contents of the file

Reply:
Instead, I error out and say you can only run saved files… if we get user feedback, we can go with a different plan but this is good for now